diagrams/VoxelSniperAbstractionClassDiagram.drawio

```diff
@@ -1 +1 @@
t=eastwest;glass=0;" parent="88" vertex="1"><mxGeometry y="52" width="270" height="8" as="geometry"/></mxCell><mxCell id="91" value="BukkitEntityType" style="swimlane;fontStyle=1;align=center;verticalAlign=top;childLayout=stackLayout;horizontal=1;startSize=26;horizontalStack=0;resizeParent=1;resizeParentMax=0;resizeLast=0;collapsible=1;marginBottom=0;glass=0;" parent="WIyWlLk6GJQsqaUBKTNV-1" vertex="1"><mxGeometry x="-1270" y="-508" width="250" height="60" as="geometry"/></mxCell><mxCell id="92" value="- entityType: org.bukkit.entity.EntityType" style="text;strokeColor=none;fillColor=none;align=left;verticalAlign=top;spacingLeft=4;spacingRight=4;overflow=hidden;rotatable=0;points=[[0,0.5],[1,0.5]];portConstraint=eastwest;glass=0;" parent="91" vertex="1"><mxGeometry y="26" width="250" height="26" as="geometry"/></mxCell><mxCell id="93" value="" style="line;strokeWidth=1;fillColor=none;align=left;verticalAlign=middle;spacingTop=-1;spacingLeft=3;spacingRight=3;rotatable=0;labelPosition=right;points=[];portConstraint=eastwest;glass=0;" parent="91" vertex="1"><mxGeometry y="52" width="250" height="8" as="geometry"/></mxCell><mxCell id="94" value="" style="endArrow=block;dashed=1;endFill=0;endSize=12;html=1;rounded=0;entryX=0.5;entryY=1;entryDx=0;entryDy=0;exitX=0.75;exitY=0;exitDx=0;exitDy=0;" parent="WIyWlLk6GJQsqaUBKTNV-1" source="87" target="91" edge="1"><mxGeometry width="160" relative="1" as="geometry"><mxPoint x="-1240" y="-566" as="sourcePoint"/><mxPoint x="-1150" y="-536" as="targetPoint"/></mxGeometry></mxCell><mxCell id="95" value="" style="endArrow=block;dashed=1;endFill=0;endSize=12;html=1;rounded=0;entryX=0.5;entryY=1;entryDx=0;entryDy=0;exitX=0.25;exitY=0;exitDx=0;exitDy=0;" parent="WIyWlLk6GJQsqaUBKTNV-1" source="87" target="88" edge="1"><mxGeometry width="160" relative="1" as="geometry"><mxPoint x="-1332.5" y="-568.9999999999995" as="sourcePoint"/><mxPoint x="-1095" y="-486" as="targetPoint"/></mxGeometry></mxCell><mxCell id="101" value="&lt;p style=&quot;margin: 0px ; margin-top: 4px ; text-align: center&quot;&gt;&lt;b&gt;VoxelMaterial&lt;/b&gt;&lt;/p&gt;&lt;hr size=&quot;1&quot;&gt;&lt;p style=&quot;margin: 0px ; margin-left: 4px&quot;&gt;+ static BLOCKS: Map&amp;lt;String, VoxelMaterial&amp;gt;&lt;/p&gt;&lt;p style=&quot;margin: 0px ; margin-left: 4px&quot;&gt;- key: String&lt;/p&gt;&lt;p style=&quot;margin: 0px ; margin-left: 4px&quot;&gt;- namespace: String&lt;/p&gt;&lt;p style=&quot;margin: 0px ; margin-left: 4px&quot;&gt;- version: Version&lt;/p&gt;&lt;p style=&quot;margin: 0px ; margin-left: 4px&quot;&gt;- static main: IVoxelSniper&lt;/p&gt;&lt;hr size=&quot;1&quot;&gt;&lt;p style=&quot;margin: 0px ; margin-left: 4px&quot;&gt;+ static setMain(IVoxelsniper main): void&lt;/p&gt;" style="verticalAlign=top;align=left;overflow=fill;fontSize=12;fontFamily=Helvetica;html=1;" vertex="1" parent="WIyWlLk6GJQsqaUBKTNV-1"><mxGeometry x="890" y="-122" width="260" height="202" as="geometry"/></mxCell></root></mxGraphModel></diagram></mxfile>
```

Comment:
Few general comments on the diagram:
- ForgeEntity is duplicated, the one that implements IEntityType should probably be called ForgeEntityType??
- I do not think we would need a setBlock(location, IBlock), since the block already has a lot of set methods?
- Also the methods in the subclasses don't seem to match
- I don't really get why VoxelMaterial needs to know the IVoxel instance?

Reply:
It needs to know the IVoxel instance to get the version on whether to return the block or null

Reply:
The setblock method on what class?
I have switched to only noting methods in the highest class they are defined, so I don't have to type down the info in 3 (or more in the future) places

Reply:
The setblock in world
it's fine to not type them, but now it's confusing, since they don't match (just remove them from the subclass)